de.tu_bs.cs.isf.e4cf.compare.data_structures/src/de/tu_bs/cs/isf/e4cf/compare/data_structures/io/reader/Visitor.java

```diff
@@ -0,0 +1,989 @@
+package de.tu_bs.cs.isf.e4cf.compare.data_structures.io.reader;
+
+import de.tu_bs.cs.isf.e4cf.compare.data_structures.impl.NodeImpl;
+import de.tu_bs.cs.isf.e4cf.compare.data_structures.interfaces.Node;
+import javafx.scene.effect.Lighting;
+
+import com.github.javaparser.ast.visitor.*;
+import com.github.javaparser.ast.body.*;
+import com.github.javaparser.ast.expr.*;
+import com.github.javaparser.ast.stmt.*;
+import com.github.javaparser.ast.type.*;
+
+import java.util.Arrays;
+
+import com.github.javaparser.JavaParser;
+import com.github.javaparser.ast.*;
+
+/**
+ * Part of the custom visitor class. This class extends VoidVisitorAdapter
+ * (https://www.javadoc.io/doc/com.github.javaparser/javaparser-core/latest/com/github/javaparser/ast/visitor/VoidVisitorAdapter.html).
+ * 
+ * @author Serkan Acar
+ * @author Pascal Blum
+ * @author Paulo Haas
+ * @author Hassan Smaoui
+ *
+ */
+public class Visitor extends VoidVisitorAdapter<Node> {
+	/**
+	 * https://www.javadoc.io/static/com.github.javaparser/javaparser-core/3.17.0/com/github/javaparser/ast/CompilationUnit.html
+	 */
+	@Override
+	public void visit(CompilationUnit n, Node arg) {
+		Node cu = VisitorUtil.Parent(n, arg);
+
+		// Imports
+		Node imports = VisitorUtil.Node(JavaNodeTypes.Import, cu);
+		int importSize = n.getImports().size();
+		VisitorUtil.AddAttribute(imports, JavaAttributesTypes.Children, String.valueOf(importSize));
+		for (int i = 0; i < importSize; i++) {

```

Comment:
Yeah, you have to write more comment above the method above critical parts such as this :P